### PR TITLE
Weighted: Add oracle enable method

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: contracts/pools/weighted/WeightedPool2Tokens.sol [hunk 1]
```diff
@@ -185,20 +188,29 @@ contract WeightedPool2Tokens is
...
+    }
+
     // Caller must be approved by the Vault's Authorizer
     function setPaused(bool paused) external authenticate {
         _setPaused(paused);
     }
 
+    function _isOwnerOnlyAction(bytes32 actionId) internal view override returns (bool) {
+        // This implementation adds support for the enableOracle action identifier.
+        return super._isOwnerOnlyAction(actionId) || actionId == getActionId(WeightedPool2Tokens.enableOracle.selector);
```
Comment (nventuro) [inline contracts/pools/weighted/WeightedPool2Tokens.sol:211]: The requirement is that governance can _always_ enable the oracle. For simplicity's sake, I thought of not letting the owner do this (which means governance can) - it seems like a weird permission for the owner regardless. If they want it on, they can always turn it on at creation.

(meaning I'd remove this override)

File: contracts/pools/weighted/WeightedPool2Tokens.sol [hunk 2]
```diff
@@ -736,7 +748,7 @@ contract WeightedPool2Tokens is
     // Oracle functions
 
     function _updateOracle(uint256 lastChangeBlock) internal {
-        if (_isOracleEnabled() && block.number > lastChangeBlock) {
+        if (isOracleEnabled() && block.number > lastChangeBlock) {
             bytes32 miscData = _miscData;
```
Comment (nventuro) [inline contracts/pools/weighted/WeightedPool2Tokens.sol:752]:
```suggestion
        bytes32 miscData = _miscData;
        bool oracleEnabled = _miscData.decodeBool(_MISC_ORACLE_ENABLED_OFFSET);
        if (oracleEnabled() && block.number > lastChangeBlock) {
```

Seems a bit silly to read _miscData twice, no? Regardless, we can ignore this now, and focus on performance once all parts are ready.

File: contracts/pools/weighted/WeightedPool2Tokens.sol [hunk 3]
```diff
@@ -185,20 +188,29 @@ contract WeightedPool2Tokens is
         emit SwapFeePercentageChanged(swapFeePercentage);
     }
 
-    function _isOracleEnabled() private view returns (bool) {
-        return _miscData.decodeBool(_MISC_ORACLE_ENABLED_OFFSET);
+    function enableOracle() external authenticate {
+        _setOracleEnabled(true);
```
Comment (nventuro) [inline contracts/pools/weighted/WeightedPool2Tokens.sol:192]:
```suggestion
        _setOracleEnabled(true);
        _cacheInvariantAndSupply();
```

File: contracts/pools/weighted/WeightedPool2Tokens.sol (hunk 2 above)
Comment (facuspagnuolo) [reply contracts/pools/weighted/WeightedPool2Tokens.sol:752]: Yea, I did this initially without taking into account gas reads/writes optimizations for this variable in particular... I have a whole plan to refactor the way we access to it entirely once the integration is finished. I wouldn't spend to much time about it rn.

File: contracts/pools/weighted/WeightedPool2Tokens.sol (hunk 3 above)
Comment (facuspagnuolo) [reply contracts/pools/weighted/WeightedPool2Tokens.sol:192]: Agreed as we discussed off-line... let's add some tests for it

**Review by nventuro — APPROVED** [review]

**Pull request merged**